src/Components/Components/src/Binding/CascadingModelBinder.cs

```diff
@@ -103,16 +100,16 @@ private void UpdateBindingInformation(string url)
         // 3) Parent has a name "parent-name"
         // Name = "parent-name.my-handler";
         // BindingContextId = <<base-relative-uri>>((<<existing-query>>&)|?)handler=my-handler
-        var name = string.IsNullOrEmpty(ParentContext?.Name) ? Name : $"{ParentContext.Name}.{Name}";
+        var name = ModelBindingContext.Combine(ParentContext, Name);
         var bindingId = string.IsNullOrEmpty(name) ? "" : GenerateBindingContextId(name);
 
         var bindingContext = _bindingContext != null &&
-            string.Equals(_bindingContext.Name, Name, StringComparison.Ordinal) &&
+            string.Equals(_bindingContext.Name, name, StringComparison.Ordinal) &&
             string.Equals(_bindingContext.BindingContextId, bindingId, StringComparison.Ordinal) ?
             _bindingContext : new ModelBindingContext(name, bindingId);
 
         // It doesn't matter that we don't check IsFixed, since the CascadingValue we are setting up will throw if the app changes.
-        if (IsFixed && _bindingContext != null && _bindingContext != bindingContext)
+        if (_bindingContext != null && _bindingContext != bindingContext)
```

Comment:
We can remove all the logic related to dealing with the binding context changing now, can't we? That was one of the main benefits to removing `IsFixed`. I don't think `UpdateBindingInformation` needs to be called on each `SetParametersAsync` or `HandleLocationChanged`, and it should be able to skip all the equality checks.

Reply:
I'm going to scope out this change of this PR, and will deal with it in a separate PR.